File: ebank/src/main/java/com/ebank/dto/TransferRequestDTO.java
package com.ebank.dto;

import com.ebank.enums.BankName;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class TransferRequestDTO {

    // For internal transfers
    private String targetAccountNumber;

    // For external transfers
    private String rib;

    private BankName bankName;

    // Common attributes
    private Double amount;

    private String description;
}

File: Ebank-Frontend/src/app/dto/transfer-request.dto.ts
import { BankName } from '../enum/bank-name';

export class TransferRequestDto {
  targetAccountNumber: string;
  rib: string;
  bankName: BankName;
  amount: number;
  description: string;

  constructor(targetAccountNumber: string, rib: string, bankName: BankName, amount: number, description: string) {
    this.targetAccountNumber = targetAccountNumber;
    this.rib = rib;
    this.bankName = bankName;
    this.amount = amount;
    this.description = description;
  }
}
